### ISSUE TITLE
Merge with lm-3/chaos ?

### ISSUE THREAD

**Issue by ams** [larsbrinkhoff/rtape#1]
I am guessing you are shoehorning it like the FIlE server or something to work with sockets? 

Wouldn’t it then make more sense to put this in lm-3/chaos (master branch)?

**Comment by larsbrinkhoff** [comment]
I won't be doing anything at all at the moment.  Just saving it away for the future.

Maybe port it over to use TCP/IP instead.

I might well contribute back if I make it work somehow.

**Comment by ams** [comment]
The code should work as is, if you have a rmt device, or — just having looked over the code — use the faking_tape variable (which is set right now).

At least if you run the kernel module with a Linux 2.0.xx kernel. :-)

**Comment by larsbrinkhoff** [comment]
I can't compile this as is.  I see these problems:

- Undefined constant: CHIOCACCEPT
- Undeclared functions: chlisten, chwaitfornotstate, chreject, tstatus, fatal, parseoptions, itoc2, checkmount, checkeof, numrec, filespace
- Incompatible pointer assignments

I'll rummage through the lm-3/chaos repository to see if there are more files I should copy.

**Comment by larsbrinkhoff** [comment]
K&R style function definitions.  Loose use of typing.

**Comment by ams** [comment]
There are a bunch of minor issues, but the code compiles (here, using gcc 6.4).   Are you compiling it in the ams/next branch or trying your own stab at it?

The chaos.h header in the rtape repository is not the one you want.  You want this: https://github.com/LM-3/chaos/blob/ams/next/h/chaos.h

**Comment by ams** [comment]
It should be noted that while there are similarities between lm-3/chaos master and ams/next, they are totally different and incompatible.  

Master is a “hacked” up version by @lisper of the original code (which lives with modifications to work on modernish systems on ams/next).

**Comment by ams** [comment]
Tried it on my test rig, talking between RTAPE and the NCP using the chtape command works.

**Comment by larsbrinkhoff** [comment]
Thanks, I'll check my chaos.h.

**Comment by larsbrinkhoff** [comment]
I had the wrong chaos.h.

I think I'd like a TCP/IP version.  And accessing a tape image file instead of a tape device.

**Comment by ams** [comment]
Makes sense, would be useful in general for other services too (thinking FILE at least)
